Generator/Sources/RegistryFile.swift

```diff
@@ -0,0 +1,140 @@
+/// Structure of an OTEL semconv registry.yaml file.
+/// All registry files are of type `attribute_group`
+/// See: https://github.com/open-telemetry/weaver/blob/main/schemas/semconv-syntax.md#attribute-group-semantic-convention
+struct RegistryFile: Codable {
+    let groups: [Group]
+}
+
+struct Group: Codable {
+    let id: String
+    let type: GroupType
+    let stability: Stability?
+    let display_name: String?
+    let brief: String?
+    let attributes: [Attribute]
+
+    enum GroupType: String, Codable {
+        case attribute_group
+        case metric
+        case span
+    }
+
+    enum Instrument: String, Codable {
+        case gauge
+        case updowncounter
+    }
+}
+
+struct Attribute: Codable {
+    let id: String
+    let type: AttributeType
+    let stability: Stability
+    let brief: String?
+    let note: String?
+    let deprecated: String?
+    let examples: [String]?
+
+    enum RequirementLevel: String, Codable {
+        case required
+    }
+
+    init(from decoder: any Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        id = try container.decode(String.self, forKey: .id)
+        if let type = try? container.decode(StandardType.self, forKey: .type) {
+            self.type = type
+        } else if let type = try? container.decode(EnumType.self, forKey: .type) {
+            self.type = type
+        } else {
+            throw DecodingError.dataCorrupted(.init(codingPath: container.codingPath, debugDescription: "Unexpected `type` value"))
+        }
+        stability = try container.decodeIfPresent(Stability.self, forKey: .stability) ?? .experimental
+        brief = try container.decodeIfPresent(String.self, forKey: .brief)
+        note = try container.decodeIfPresent(String.self, forKey: .note)
+        deprecated = try container.decodeIfPresent(String.self, forKey: .deprecated)
+        if !container.contains(.examples) {
+            examples = nil
+        } else if let example = try? container.decode(String.self, forKey: .examples) {
+            examples = [example]
+        } else {
+            let examples = try? container.decode([String].self, forKey: .examples)
+            self.examples = examples
+        }
+    }
+
+    func encode(to encoder: any Encoder) throws {
```

Comment:
Do we need the `Encodable` conformance here?

Reply:
No, we don't. I've removed encodable conformance in https://github.com/swift-otel/swift-otel-semantic-conventions/pull/2/commits/c2485fda8087bcd60dc511f14bcd3660fb6404b6